Common/Core/RecoDecay.h

```diff
@@ -584,12 +582,12 @@ class RecoDecay
     if (sign) {
       *sign = sgn;
     }
-    while (particleMother.has_mother0()) {
+    while (particleMother.has_mothers()) {
       if (depthMax > -1 && -stage >= depthMax) { // Maximum depth has been reached.
         return -1;
       }
-      auto indexMotherTmp = particleMother.mother0Id();
-      particleMother = particlesMC.iteratorAt(indexMotherTmp);
+      particleMother = particleMother.template mothers_first_as<typename std::decay_t<T>::parent_t>(); // get mother 0
```

Comment:
Is this really the simplest way to get the first mother?

Reply:
This was suggested by @aalkin, I don't think there are easier ways to do it

Reply:
All these is required because we need to call a dependent template inside a templated function, in `process()` one would not need the `template` keyword and type calculation, of course.